src/platform_impl/macos/app_delegate.rs

```diff
@@ -79,3 +106,45 @@ extern "C" fn did_resign_active(this: &Object, _: Sel, _: id) {
     }
     trace!("Completed `applicationDidResignActive`");
 }
+
+extern "C" fn application_open_file(_this: &Object, _: Sel, _sender: id, filename: id) -> BOOL {
+    trace!("Triggered `application:openFile:`");
+    let mut filenames_vec = Vec::with_capacity(1);
+    let string = ns_string_to_rust(filename);
+    filenames_vec.push(string.into());
+    let event = Event::OpenFiles(filenames_vec);
+    AppState::queue_event(EventWrapper::StaticEvent(event));
+    trace!("Completed `application:openFile:`");
+
+    // Return true to indicate to the OS that the file type is supported.
+    // (If the filetype turns out not to be supported, it's the application's
+    // responsibility to inform the user)
+    YES
+}
+
+extern "C" fn application_open_files(_this: &Object, _: Sel, _sender: id, filenames: id) {
+    use cocoa::foundation::NSArray;
+
+    #[allow(non_upper_case_globals)]
+    const NSApplicationDelegateReplySuccess: i32 = 0;
```

Comment:
Please put this constant at the top of the file. Someone else might want to use it and we don't want to redefine it a billion times.

Also there's a reason why you got a warning; please fix it instead of just silencing it.

Reply:
> Please put this constant at the top of the file.

Right away.

> Also there's a reason why you got a warning; please fix it instead of just silencing it.

I understand your concern but I feel like the tone was a bit uncalled for. The reason why I silenced the warning was that this constant gets to have the exact name that it was given in appkit. This is consistent with how types and functions are exposed by the `cocoa` crate.

Reply:
@chrisduerr you're being rough throughout this PR, and it looks like you ignored @ArturKovacs here instead of apologizing. Even if that wasn't your intention, it's still how @ArturKovacs feels, and that's something important to address.

I know open source can be stressful, so please don't push yourself too hard.

Reply:
In an attempt to help resolving this, I'd like to add that I'm not upset with you @chrisduerr. I did feel like your approach was not very considerate but I'm willing to move on.